### PR TITLE
AppFlowy build tasks for Android platform

### PR DESCRIPTION
**Pull request by rileyhawk1417**
### Feature Preview

- This PR is related to [AppFlowy Editor](https://github.com/AppFlowy-IO/appflowy-editor/issues/68), so its in preparation for when AppFlowy Editor supports the mobile version. The PR just has the build tasks to make it run on android.
- There is one issue with it working on Windows platform. The rust backend does not compile for android, otherwise there's nothing wrong with the dart(frontend) part.

---

#### PR Checklist

- [x] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [ ] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/.vscode/launch.json [hunk 1]
```diff
@@ -43,6 +43,16 @@
             },
             "cwd": "${workspaceRoot}/appflowy_flutter"
         },
+        {
+            "name": "AF-iOS: Build Dart Only",
```
Comment (rileyhawk1417) [inline frontend/.vscode/launch.json:47]: Added this task just for people who want to run AppFlowy, without rebuilding the backend

**Comment by rileyhawk1417** [comment]
@LucasXu0 I had put instructions in a README inside the android folder. Haven't tested it on MacOS yet, but am sure the setup is similar to that of Linux.

> @rileyhawk1417 Can you provide more details on the instructions?
> 
> I encountered this error.
> 
> ```
> error: Could not find any NDK.
> note: Set the environment ANDROID_NDK_HOME to your NDK installation's root directory,
> or install the NDK using Android Studio.
> ```

The error is related to the Android NDK not being found, its what is used to compile the rust backend for android.

**Comment by AkechiShiro** [comment]
@LucasXu0 I believe you should look into the readme that was modified in the commits, there seems TL be a list of requirements where Android NDK is mentioned along other stuff, something called cargo-ndk as well.

If I understand correctly, the app is written with a blend of Flutter (Dart) and Rust as well, I don't understand just which kind of part is written in Rust (what is that part achieving ?).

I've stumbled on this project and I'm interesting in having it working on Android, just keeping an eye on this PR since this brings good progress towards building a mobile application.

@rileyhawk1417 Is it recommended in order to test this PR, to use VSCode ? I saw some files related to it.

**Comment by rileyhawk1417** [comment]
@AkechiShiro there's an [article](https://blog.appflowy.io/tech-design-flutter-rust/) explaining about how flutter & rust work together in the app. As for the testing part of VSCode well its not necessary but it just makes things easier for development, you can run the VSCode tasks individually on a separate terminal.

File: frontend/.vscode/tasks.json [hunk 2]
```diff
@@ -84,6 +84,34 @@
         "cwd": "${workspaceFolder}"
       }
     },
+    {
+      "label": "AF: Clean + Rebuild All (Android)",
```
Comment (Xazin) [inline frontend/.vscode/tasks.json:88]: I think it's easier to see if you're running the right task, if the label include critical information at the beginning.
```suggestion
      "label": "AF-Android: Clean + Rebuild All",
```

File: frontend/.vscode/tasks.json [hunk 3]
```diff
@@ -84,6 +84,34 @@
...
+        "AF: Generate Freezed Files",
+        "AF: Generate Svg Files"
+      ],
+      "presentation": {
+        "reveal": "always",
+        "panel": "new"
+      }
+    },
+    {
+      "label": "AF: Build Appflowy Core For Android",
```
Comment (Xazin) [inline frontend/.vscode/tasks.json:107]:
```suggestion
      "label": "AF-Android: Build Appflowy Core",
```

**Comment by Xazin** [comment]
Do you have a take on this PR @LucasXu0 ?

**Comment by rileyhawk1417** [comment]
Updated the debug labels & added a .gitignore for the android build files.

**Comment by rileyhawk1417** [comment]
It did fail to build the flowy test crate with this error:
```
warning: field `storage_service` is never read
  --> flowy-document2/src/manager.rs:35:3
   |
29 | pub struct DocumentManager {
   |            --------------- field in this struct
...
35 |   storage_service: Weak<dyn FileStorageService>,
   |   ^^^^^^^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

   Compiling flowy-core v0.1.0 (/home/riley/Documents/Github_Repos/appFlowy/app/appflowy/frontend/rust-lib/flowy-core)
warning: `flowy-document2` (lib) generated 1 warning
   Compiling flowy-test v0.1.0 (/home/riley/Documents/Github_Repos/appFlowy/app/appflowy/frontend/rust-lib/flowy-test)
error[E0599]: no method named `get_cloud_service` found for struct `Arc<DocumentManager>` in the current scope
  --> flowy-test/src/lib.rs:90:47
   |
90 |     let cloud_service = self.document_manager.get_cloud_service().clone();
   |                                               ^^^^^^^^^^^^^^^^^ method not found in `Arc<DocumentManager>`
```

I did pull from the latest code so maybe something is missing?

**Comment by rileyhawk1417** [comment]
Closing this PR as its been solved by this [PR](https://github.com/AppFlowy-IO/AppFlowy/pull/3713)

**Pull request closed without merging**